src/edu/washington/escience/myria/operator/SamplingDistribution.java

```diff
@@ -0,0 +1,300 @@
+package edu.washington.escience.myria.operator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.column.Column;
+import edu.washington.escience.myria.column.builder.IntColumnBuilder;
+import edu.washington.escience.myria.column.builder.StringColumnBuilder;
+import edu.washington.escience.myria.storage.TupleBatch;
+
+public class SamplingDistribution extends UnaryOperator {
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  /** The output schema. */
+  private static final Schema SCHEMA = Schema.of(
+      ImmutableList.of(Type.INT_TYPE, Type.INT_TYPE, Type.INT_TYPE, Type.STRING_TYPE),
+      ImmutableList.of("WorkerID", "StreamSize", "SampleSize", "SampleType"));
+
+  /** Total number of tuples to sample. */
+  private int sampleSize;
+
+  /** True if using a percentage instead of a specific tuple count. */
+  private boolean isPercentageSample = false;
+
+  /** Percentage of total tuples to sample. */
+  private float samplePercentage;
+
+  /** The type of sampling to perform. Currently supports 'WR' and 'WoR'. */
+  private final String sampleType;
```

Comment:
I think this should be an enum (looks like Jackson supports serializing enums as strings). There can be compatibility issues with binary serialization of enums as integer values, but I think serializing them as strings should be fine.